src/components/include/utils/threads/thread.h

```diff
@@ -220,6 +235,34 @@ class Thread {
   sync_primitives::ConditionalVariable state_cond_;
 
  private:
+  void SetState(const ThreadState thread_state) {
+    thread_state_ = thread_state;
+  }
+
+  void SwitchOnState(const ThreadState thread_state) {
+    if (IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  void SwitchOffState(const ThreadState thread_state) {
+    if (!IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  bool IsSwitchOnState(const ThreadState thread_state) const {
+    return thread_state & thread_state_;
```

Comment:
I would propose to rename `thread_state_` to `current_thread_state_` so it could be more obvious.

Reply:
@YarikMamykin 
Maybe, but we don’t have another state of the stream, we only save the current state.
I think current_thread_state_ implies the presence of other states that are not in the thread, according to this maybe thread_state_ is more short, obvious and clear ? 
Is aggree ?

Reply:
@SNiukalov ok, I agree.